### ISSUE TITLE
HttpRequest.headers fixer produces unsupported assigments in tests

### ISSUE THREAD

**Issue by browniebroke** [adamchainz/django-upgrade#74]
### Python Version

3.8

### Django Version

3.1

### Package Version

1.3.1

### Description

I ran django-upgrade on our codebase and it replaced some places where we were setting `request.META` to a fake requests created by [`RequestFactory`](https://docs.djangoproject.com/en/stable/topics/testing/advanced/#django.test.RequestFactory) in our tests, which isn't correct.

I guess `django-upgrade` shouldn't change these if they are on the left hand side of an assignment.

Here is an example:

```diff
  request = RequestFactory().get('/')
- request.META['HTTP_USER_AGENT'] = user_agent
+ request.headers['User-Agent'] = user_agent
```

When running tests, it causes: `TypeError: 'HttpHeaders' object does not support item assignment`. 

### known workarounds

1. Rename the variable to something else than `request`, e.g. `req`
2. In this specific example, I was able to avoid django-upgrade to touch our code if I rewrite it with:

   ```diff
   - request = RequestFactory().get('/')
   - request.META['HTTP_USER_AGENT'] = user_agent 
   + request = RequestFactory(HTTP_USER_AGENT=user_agent).get('/')
   ```

**Comment by adamchainz** [comment]
Ah, I did not know that one could assign to request.META in tests

We can heuristically avoid touching test requests by checking if the node's parent is a function whose name starts with `test_`...

(Also assigning to `request.headers` in tests sounds like a useful feature for Django to add, if you want to find/make a ticket there.)

**Comment by adamchainz** [comment]
> I guess `django-upgrade` shouldn't change these if they are on the left hand side of an assignment.

I missed this. This is easy to do.

**Comment by adamchainz** [comment]
Fixed in #75, relaesed in version 1.3.2.

**Comment by browniebroke** [comment]
Nice! 🚀
